core/textile/bucket_factory.go

```diff
@@ -259,3 +266,63 @@ func (tc *textileClient) JoinBucket(ctx context.Context, slug string, ti *domain
 
 	return true, nil
 }
+
+func (tc *textileClient) getContextPubKey(ctx context.Context) (thread.PubKey, error) {
+	kc := keychain.New(tc.store)
+	rk, _, err := kc.GetRootKeyPairInLibP2PFormat()
+	if err != nil {
+		return nil, err
+	}
+
+	tok, _ := thread.TokenFromContext(ctx)
+	pk, err := tok.Validate(rk)
+	return pk, nil
+}
+
+func (tc *textileClient) saveInviteRecord(ctx context.Context, slug string) error {
+	bctx, dbID, err := tc.initBucketThreadMetaCollection(ctx, slug, bucketInviteCollectionName, &BucketInvite{})
+
+	pk, err := tc.getContextPubKey(ctx)
```

Comment:
Not sure if this is the public key we want, as this will be the public key of the current user and not the one that receives the invitation I think.

Reply:
The one that receives it is being saved on the `JoinThread` call.  I'm saving this one here as the sender pub key so maybe it's just a matter of calling the field `SenderPubKey` or something. I don't think our scoping called the need for this specifically but just saved it while I was doing the saves here.